cmd/ci-operator/main.go

```diff
@@ -1483,3 +1485,41 @@ func (o *options) getResolverInfo(jobSpec *api.JobSpec) *load.ResolverInfo {
 	}
 	return info
 }
+
+func monitorNamespace(ctx context.Context, cancel func(), namespace string, client coreclientset.NamespaceInterface) {
+	for {
+		watcher, err := client.Watch(meta.ListOptions{
+			TypeMeta:      meta.TypeMeta{},
+			FieldSelector: fields.Set{"metadata.name": namespace}.AsSelector().String(),
+			Watch:         true,
+		})
+		if err != nil {
+			log.Printf("Could not start a watch on our test namespace... (details; %v)", err)
+			cancel()
+			return
+		}
+		for {
+			select {
+			case <-ctx.Done():
+				// we're done operating anyway
+				return
+			case event, ok := <-watcher.ResultChan():
+				if !ok {
+					continue
+				}
+				ns, ok := event.Object.(*coreapi.Namespace)
+				if !ok {
+					continue
+				}
+				if ns.Name != namespace {
+					continue
+				}
+				if ns.DeletionTimestamp != nil {
+					log.Print("The namespace in which this test is executing has been deleted, cancelling the test...")
```

Comment:
Under what circumstances can this happen?

Reply:
They're all weird edge cases or have to do with manual action or other nonsense. We just want to print this nice message instead of falling over due to "RBAC"